### DISCUSSION TITLE
UCB CS189: Introduction to Machine Learning - CS自学指南

### DISCUSSION THREAD

**Discussion by giscus** [PKUFlyingPig/cs-self-learning#185] [Announcements]
# UCB CS189: Introduction to Machine Learning - CS自学指南

CS自学指南

https://csdiy.wiki/%E6%9C%BA%E5%99%A8%E5%AD%A6%E4%B9%A0/CS189/

**Comment by Mike-Sagiri** [comment]
好像2022年春季的课程不再开源autograder和notes了，所以有没有哪位大佬能够提供notes呀

**Reply by Mike-Sagiri** [reply]
Note自己后来找到了。在课程***resource***里。但是作业gradescope的确没有给出链接了

**Reply by soondubu137** [reply]
请问你找到带autograder的作业了吗？

**Reply by Mike-Sagiri** [reply]
我不知道***homework***栏目下的压缩文件作业里是否带有autograder，但是gradescope的自动评分是不对外开放的了。

**Reply by YangHuaLuoJinZiGuiTi** [reply]
CS189 春季的project后面用kaggle进行评分， 可以去做。
在kaggle中搜索CS189后按照时间排序就能找到对应年份的每个作业competition

**Reply by ChenxiZhang168** [reply]
最新的2023AU的是E73744

**Reply by ChenxiZhang168** [reply]
哪个大佬有往年的嘛 没锁住的

**Reply by ChenxiZhang168** [reply]
能发一下kaggle的链接嘛

**Reply by Mamihla** [reply]
作业的pdf文档里一般都会给出对应kaggle链接

**Reply by Mamihla** [reply]
春季的是这样的 直接提交csv文件到对应kaggle比赛就能打分 ，至于秋季的就不清楚了

**Reply by Lubenwei-nb123** [reply]
同求往年没锁的课程码

**Comment by AI-Kyo-er** [comment]
找到一个2023 Spring 的 该课程，https://people.eecs.berkeley.edu/~jrs/189/，游客可以访问屏幕录制，好像比油管上的视频更加清晰

**Reply by wsehjk** [reply]
请问有23 spring的autograder 的entrycode吗

**Reply by AI-Kyo-er** [reply]
肯定没有吧，这个应该是内部人员才有的。

**Comment by Mamihla** [comment]
spring23的视频：https://www.youtube.com/playlist?list=PLppfe3aWPcD4fthieedXTrE3kQz8rvtJt （偷偷问这是能上传油管的吗🤷‍

**Comment by ChenxiZhang168** [comment]
有无大佬有之前的public 的gradescope的激活码啊？最新的2023AU的是E73744，但是有due date，due date之后好像就锁了

**Comment by rinevard** [comment]
sp和fa的版本在内容上有挺大的不同，fa的版本偏理论，难度也较大，具体区别可以参考这个贴子:

https://www.reddit.com/r/berkeley/comments/1b67zhw/thoughts_on_fall_cs_189_compared_to_spring_189/

不过我注意到fa版本的作业似乎没有完全开源，官网中只给出了部分作业的文件，而且也没有答案，如果有同学知道在哪里可以找到这些文件可以分享一下～

**Reply by sarendis56** [reply]
推荐Spring的版本，除了手写作业答案都是公开的，视频可以在B站上找到，Coding作业可以在官网下载并在Kaggle上提交（接受late submission），lecture notes (LaTeX) 可以在教授的网站上找到，https://people.eecs.berkeley.edu/~jrs/ 翻到professing并选择你看的学期

**Reply by sarendis56** [reply]
本来是发在帖子下面的不小心点成回复了不好意思：）

**Reply by CheeseSilly** [reply]
感谢，上面的网站真的很有帮助

**Reply by amn765** [reply]
请问spring的官网是什么呢，wayback machine上面好像只能找到fall的
想要做coding作业，但看别人似乎是在ipynb上面做的，教授的网站上没有

**Reply by CheeseSilly** [reply]
spring的官网似乎就是上面的，我也没有发现其他的。作业确实用Jupyter写的，也许可以先学Data100再学这个，应该会很有帮助。

**Comment by zhiquanlao** [comment]
好像kaggle不能late submit了

**Comment by weathyII** [comment]
大佬们，求182的

**Comment by Vitoom** [comment]
最新的课程视频 2024 Fall https://berkeley-haas.hosted.panopto.com/Panopto/Pages/Sessions/List.aspx#folderID=%22380bd203-98f7-4a83-946e-b1d401302c37%22

**Reply by Vitoom** [reply]
这门课教材用的还是Bishop的新书 https://www.bishopbook.com/

**Comment by Elexs126** [comment]
看了一下这些课在理论方面还是相对较弱，因为这领域技术迭代比较快，而且学习人员背景复杂，这些教学思路倒也可以。
但如果对机器学习的数学原理想获得更透彻理解，或者想把各种行为和算法统一在一个框架上，可以看一下图宾根的 2025  Probabilistic ML 。课程的大概参考书是Murphy 的PML，但具体的内容比书新很多，同时也有大量的code。

**Reply by Shang-fei** [reply]
您好，请问如果研究生搞cv或者llm这些方向，需要对Probabilistic ML有所了解嘛，本人马上研一。

**Reply by Elexs126** [reply]
> 您好，请问如果研究生搞cv或者llm这些方向，需要对Probabilistic ML有所了解嘛，本人马上研一。

根据需求去学习，而不是学习去收集

**Comment by uit867r68tli8** [comment]
在学24fall，课程网站突然就变成23fall的了，请问有人存了24fall的homework,discussion吗

**Reply by RisingUppercut** [reply]
https://web.archive.org/web/20250118005857/https://eecs189.org/     
这个可以，话说我也在学，你现有hw的solution吗，24fall的有kaggle提交的吗

**Reply by uit867r68tli8** [reply]
感谢，好像只有spring的有kaggle提交

**Comment by RisingUppercut** [comment]
今年25fall的课程改了不少，很多传统的内容都没讲，加入了不少深度学习的内容。

**Reply by LyrineYang** [reply]
似乎今年的课程改革了,老师和slide都很新,有任何25fall的课程资源吗?想试着听听看

**Comment by RisingUppercut** [comment]
我能搞到recording，要一起学吗


---- 回复的原邮件 ----
发件人  Junqi ***@***.***> 日期  2025年09月05日 18:34 收件人  ***@***.***> 抄送至  David ***@***.***>***@***.***> 主题  Re: [PKUFlyingPig/cs-self-learning] UCB CS189: Introduction to Machine Learning - CS自学指南 (Discussion #185)  

似乎今年的课程改革了,老师和slide都很新,有任何25fall的课程资源吗?想试着听听看

**Comment by RisingUppercut** [comment]
1062109984，建了个群，可以一起


---- 回复的原邮件 ----
发件人  Junqi ***@***.***> 日期  2025年09月05日 18:34 收件人  ***@***.***> 抄送至  David ***@***.***>***@***.***> 主题  Re: [PKUFlyingPig/cs-self-learning] UCB CS189: Introduction to Machine Learning - CS自学指南 (Discussion #185)  

似乎今年的课程改革了,老师和slide都很新,有任何25fall的课程资源吗?想试着听听看

**Reply by phucthinhdoan** [reply]
我怎样才能加入这个小组？

**Comment by RisingUppercut** [comment]
qq群，the group number of QQ，an app of china


---- 回复的原邮件 ----
发件人  ***@***.***> 日期  2025年09月13日 15:36 收件人  ***@***.***> 抄送至  David ***@***.***>***@***.***> 主题  Re: [PKUFlyingPig/cs-self-learning] UCB CS189: Introduction to Machine Learning - CS自学指南 (Discussion #185)  

我怎样才能加入这个小组？

**Reply by phucthinhdoan** [reply]
还有其他方式观看视频吗？我也想进群。你们有不和谐吗？

**Comment by RisingUppercut** [comment]
进群直接加就行


---- 回复的原邮件 ----
发件人  ***@***.***> 日期  2025年09月13日 17:54 收件人  ***@***.***> 抄送至  David ***@***.***>***@***.***> 主题  Re: [PKUFlyingPig/cs-self-learning] UCB CS189: Introduction to Machine Learning - CS自学指南 (Discussion #185)  

还有其他方式观看视频吗？我也想进群。你们有不和谐吗？

**Comment by vixitor** [comment]
https://github.com/vixitor/berkeleycs189 sp25我在伯克利上过这门课这是当时的笔记可以参考一下，个人感觉这门课难度不是特别大，春季学期来讲，700人的考试最后考了20名喜提A+，这门课最主要的作用对我来说是了解了很多机器学习的概念，但是和现在的ai研究差别还是很大的，如果能力够的话可以直接上神经网络的课，这个课很多都是传统机器学习，神经网络只涉及一点点，然后数学线代概率期望之类的东西比较多

**Reply by akseven6221** [reply]
现在不一样了，25fa一半都是深度学习的内容，跟现在的大模型挺贴近了

**Comment by cy-Yin** [comment]
我写的 Spring 2024 版本的所有资料，包括 Discussions, Homework 和 Exams 都在
[UCBerkeley-CS189-Sp24-Machine-Learning](https://github.com/cy-Yin/UCBerkeley-CS189-Sp24-Machine-Learning)
欢迎讨论

2025 以及之前的 spring 学期的 CS189 都是 [Jonathan Shewchuk](https://people.eecs.berkeley.edu/~jrs/) 教授讲的，内容以及作业都差不多，偏向传统的 Machine Learning 带一部分 Deep Learning 的内容，Fall2025 的版本改为更加贴近 Deep Learning 的形式，基本不涉及传统 Machine Learning ，数学推导也少了很多，如果有想要打下扎实的 Machine Learning 基础的同学建议学 spring22-25 的版本

**Reply by futao-byte** [reply]
太感谢了👍

**Comment by EnzoDing-rgb** [comment]
有没有人备份一下26sp的全部资料

**Comment by futao-byte** [comment]
花两周左右把讲义自己推了一遍，这个讲义写的不错，有的地方跳了步骤需要查阅书籍或者问ai，感觉把之前机器学习没学懂的地方基本领悟了，明天准备自己把final做一遍。

**Reply by futao-byte** [reply]
很谢谢cy-Yin上面那位的资料

**Reply by moyu222** [reply]
哥，看到你也学了cmu的那个10-414/714，这两个课不冲突吧，打算都上一下🤔

**Reply by futao-byte** [reply]
感觉cmu那个也很好，那个我学了一半了，这门我学的24年的，现在还在做这门作业，作业有点难
